### PR TITLE
Move all diagnostics writing on the same level

### PR DESCRIPTION
**Pull request by reinsch82**
This means we reuse the "diagnostics" property from the CC model. This makes writing the diagnostics on gradle/gradle easier.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

File: platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/cc/impl/problems/JsonModelWriter.kt [hunk 1]
```diff
@@ -18,20 +18,17 @@ package org.gradle.internal.cc.impl.problems
 
 class JsonModelWriter(val modelWriter: JsonWriter) {
     fun beginModel() {
-        with(modelWriter) {
+        modelWriter.run {
```
Comment (donat) [inline platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/cc/impl/problems/JsonModelWriter.kt:21]: ❓ This is probably not necessary; why did you change that?

File: platforms/ide/problems/src/main/kotlin/org/gradle/problems/internal/impl/DefaultProblemsReportCreator.kt [hunk 2]
```diff
@@ -65,12 +66,17 @@ class DefaultProblemsReportCreator(
...
-                    property("totalProblemCount", problemCount.toString())
-                    buildNameProvider.buildName()?.let { property("buildName", it) }
-                    property("requestedTasks", taskNames.joinToString(" "))
-                    property("documentationLink", DocumentationRegistry().getDocumentationFor("problem-report"))
-                    property("documentationLinkCaption", "Problem report")
+                    property("problemsReport") {
+                        jsonObject {
+                            property("totalProblemCount") {
+                                write(problemCount.toString())
+                            }
```
Comment (donat) [inline platforms/ide/problems/src/main/kotlin/org/gradle/problems/internal/impl/DefaultProblemsReportCreator.kt:73]: Consistency. Either

```suggestion
                            property("totalProblemCount") { write(problemCount.toString()) }
```

or add newlines to line 74.

**Review by donat — APPROVED** [review]

File: platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/cc/impl/problems/JsonModelWriter.kt (hunk 1 above)
Comment (reinsch82) [reply platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/cc/impl/problems/JsonModelWriter.kt:21]: As far as I can tell, `run` is used more consistently in our code base than `with`.

**Comment by reinsch82** [comment]
@bot-gradle test this
